docs/networking/harvester-network.md

```diff
@@ -106,3 +106,32 @@ To create a new untagged network, go to the **Networks > VM Networks** page and
 Starting from Harvester v1.1.2, Harvester supports updating and deleting VM networks. Make sure to stop all affected VMs before updating or deleting VM networks.
 
 :::
+
+## About Route Connectivity
+
+![](/img/v1.3/networking/route-connectivity.png)
+
+There are four possible cases for the **Route Connectivity** for each VM Network:
+
+- `Active`, meaning the connectivity between the VM Network and Harvester hosts via the configured gateway is confirmed.
+- `Dhcp failed`, meaning Harvester cannot get the route information via DHCP, therefore it cannot confirm the connectivity between the VM Network and Harvester hosts. Please ensure the DHCP server is configured appropriately and is L2-reachable (or routable if a DHCP relay server is provided in the target network). Otherwise, please provide the gateway IP address directly during the VM Network creation.
+- `Ping failed`, meaning Harvester is unable to send ICMP Echo Request packets. This rarely happens.
+- `Inactive`, meaning such a VM Network is not reachable (or reachable but packet loss is greater than 20%) from Harvester hosts. Please ensure the gateway is configured appropriately and is reachable via the management network where the Harvester nodes live.
+
+:::info important
+
+For the [VM load balancer feature](./loadbalancer#vm-load-balancer) to work, the VM network must be `Active` in terms of route connectivity.
+
+:::
+
+Behind the scenes, the Harvester network controller checks the connectivity of each VM Network. Connectivity means whether the target VM Network is reachable (via router(s) if necessary) from the Harvester node. The check is essential because it indicates that such a VM Network is suitable for running workloads that require connections to the Harvester node, especially the control plane. For instance, the Harvester cloud provider that is running in the guest cluster needs to access the underlying Harvester/Kubernetes APIs to be able to calculate the node topology and provide the load balancer feature.
+
+To check the connectivity, the gateway IP address is of interest to the Harvester network controller. Such information could be absent during  VM Network creation. However, it's still possible to get it if a DHCP server is running on the target VM Network and configured with the gateway information. If the user actively provides the gateway information during network creation, the network controller happily accepts it. Otherwise, the network controller will create a helper job on the target network that acts as a DHCP client to get the gateway information. With the gateway IP address in mind, the network controller then sends ICMP Echo Request packets from the management network to the gateway and waits for responses.
```

Comment:
A question, please double check:

The helper job is set with additional network annotation, which in turn creates the 2nd interface in the job POD that connects to VM network; like the storage network, LH PODs get an [additional NIC & IP](https://docs.harvesterhci.io/v1.3/advanced/storagenetwork#step-3), all the related DHCP/ping are done on this interface. Not from the management network.

```
job.Spec.Template.ObjectMeta.Annotations[cniv1.NetworkAttachmentAnnot] = selectedNetworks
```

Reply:
I've confirmed that the DHCP client runs on the target VM Network according to the pod annotation. On the other hand, the `pingGW` call is run directly inside the manager pod instead of the helper pod, so it's apparently issued from the management network:

- Initially https://github.com/harvester/network-controller-harvester/blob/ef2e5d13f0e3a37e833e6b23328b83832319d90e/pkg/controller/manager/nad/controller.go#L387
- Periodically https://github.com/harvester/network-controller-harvester/blob/ef2e5d13f0e3a37e833e6b23328b83832319d90e/pkg/controller/manager/nad/controller.go#L360

Besides, it would be useless to check the connectivity between the target VM Network and the management network just by pinging the gateway from a random IP address on the target VM network. Please correct me if I'm wrong. Thank you.

Reply:
Per the discussion and investigation:

It looks the connectivity between mgmt network and VM network is mostly related to REK2 downstream cluster scenairo. For other general cases, those two networks are blocked.

The ping detection looks to be better as an option for VM network which is enabled by the user, the user knows he need to make sure the infrastructure layer connects them.

Let's add detailed background and connect with rancher integratio part if possible, thanks.